### PR TITLE
Adding support for additional-sidebar-resources.json content files

### PR DESCRIPTION
**Pull request by ashleemboyer**
## 🔗 Relevant links

- Asana tasks 🎟️ 
  - [[Sidebar] Make Resources section customizable by product](https://app.asana.com/0/1202097197789424/1202661920988657/f)
  - [[polish] Remove 'All Tutorials' link from sidebar Resources](https://app.asana.com/0/1202097197789424/1202709301226792/f)

## 🗒️ What

- Updates `generateResourcesNavItems` to:
  - Handle generating a default set of links
  - Handle customizing the links by a product slug
  - Allow additional items to be appended to the list
- Removes `"All Tutorials"` link from Sidebar Resources section

## 🤷 Why

See the context for this approach [in Slack](https://hashicorp.slack.com/archives/C03A8CW77FU/p1659577575329129).

## 🧪 Testing

Other than the removal of the `"All Tutorials"` link, there should be no changes in the Sidebar Resources section of the following pages:

- [ ] [/vault](https://dev-portal-git-ambabstract-sidebar-resources-hashicorp.vercel.app/vault)
- [ ] [/vault/docs](https://dev-portal-git-ambabstract-sidebar-resources-hashicorp.vercel.app/vault/docs)
- [ ] [/vault/api-docs](https://dev-portal-git-ambabstract-sidebar-resources-hashicorp.vercel.app/vault/api-docs)
- [ ] [/vault/tutorials](https://dev-portal-git-ambabstract-sidebar-resources-hashicorp.vercel.app/vault/tutorials)
- [ ] [/vault/downloads](https://dev-portal-git-ambabstract-sidebar-resources-hashicorp.vercel.app/vault/downloads)
- [ ] [/waypoint](https://dev-portal-git-ambabstract-sidebar-resources-hashicorp.vercel.app/waypoint)
- [ ] [/waypoint/docs](https://dev-portal-git-ambabstract-sidebar-resources-hashicorp.vercel.app/waypoint/docs)
- [ ] [/waypoint/commands](https://dev-portal-git-ambabstract-sidebar-resources-hashicorp.vercel.app/waypoint/commands)
- [ ] [/waypoint/plugins](https://dev-portal-git-ambabstract-sidebar-resources-hashicorp.vercel.app/waypoint/plugins)
- [ ] [/waypoint/tutorials](https://dev-portal-git-ambabstract-sidebar-resources-hashicorp.vercel.app/waypoint/tutorials)
- [ ] [/waypoint/downloads](https://dev-portal-git-ambabstract-sidebar-resources-hashicorp.vercel.app/waypoint/downloads)

### CODE REVIEW

File: src/components/sidebar/helpers/generate-resources-nav-items.ts [hunk 1]
```diff
@@ -1,70 +1,47 @@
...
+		} catch {
+			developmentToast({
+				color: ToastColor.critical,
+				title: 'Error in `generateResourcesNavItems`',
+				description: `Could not load "sidebar-resources.json" content file for "${productSlug}". Rendering default Resources items instead.`,
+			})
+		}
+	}
+
+	return [{ heading: 'Resources' }, ...resourceItems]
```
Comment (ashleemboyer) [inline src/components/sidebar/helpers/generate-resources-nav-items.ts:44]: 🗒️ Note to self: update this to set `resourceItems` to `DEFAULT_ITEMS` if `data.items` is empty/falsy

Comment (ashleemboyer) [reply src/components/sidebar/helpers/generate-resources-nav-items.ts]: Irrelevant now 😊

File: src/components/sidebar/helpers/generate-resources-nav-items.ts [hunk 2]
```diff
@@ -1,29 +1,23 @@
 import { ProductSlug } from 'types/products'
 
-const DEFAULT_COMMUNITY_FORUM_LINK = 'https://discuss.hashicorp.com/'
+const DEFAULT_COMMUNITIY_FORUM_LINK = 'https://discuss.hashicorp.com/'
```
Comment (kendallstrautman) [inline src/components/sidebar/helpers/generate-resources-nav-items.ts:3]:
```suggestion
const DEFAULT_COMMUNITY_FORUM_LINK = 'https://discuss.hashicorp.com/'
```
re: @tunzor's point on the typo

File: src/content/waypoint/sidebar-resources.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+	"items": [
+		{
+			"title": "All Tutorials",
+			"href": "https://learn.hashicorp.com/"
```
[the remark this answers is not in the document]
Comment (ashleemboyer) [reply src/content/waypoint/sidebar-resources.json:5]: Oh oops! I meant to delete this. Good call out 😁

**Review by kendallstrautman — APPROVED** [review]

**Pull request merged**